contracts/tokens/SBT.sol

```diff
@@ -0,0 +1,191 @@
+     */
+    function __SBT_init(string memory name_, string memory symbol_) internal onlyInitializing {
+        _name = name_;
+        _symbol = symbol_;
+    }
+
+    /**
+     * @notice The function to return the name of the contract
+     * @return the name of the contract
+     */
+    function name() public view virtual override returns (string memory) {
+        return _name;
+    }
+
+    /**
+     * @notice The function to return the symbol of the contract
+     * @return the symbol of the contract
+     */
+    function symbol() public view virtual override returns (string memory) {
+        return _symbol;
+    }
+
+    /**
+     * @notice The function to check the existence of the token
+     * @param tokenId_ the token to check
+     * @return true if `tokenId_` exists, false otherwise
+     */
+    function tokenExists(uint256 tokenId_) public view override returns (bool) {
+        return ownerOf(tokenId_) != address(0);
+    }
+
+    /**
+     * @notice The function to get the balance of the user
+     * @param owner_ the user to get the balance of
+     * @return the user's balance
+     */
+    function balanceOf(address owner_) public view override returns (uint256) {
+        return _balances[owner_].length();
+    }
+
+    /**
+     * @notice The function to get a user's token by its ordinal id
+     * @param owner_ the user to get the token of
+     * @param index_ the id of the token in the user's array
+     */
+    function tokenOf(address owner_, uint256 index_) public view override returns (uint256) {
+        return _balances[owner_].at(index_);
+    }
+
+    /**
+     * @notice The function to get ALL the tokens of a user. Be careful, O(n) complexity
+     * @param owner_ the user to get the tokens of
+     * @return the array of tokens the user owns
+     */
+    function tokensOf(address owner_) public view override returns (uint256[] memory) {
+        return _balances[owner_].values();
+    }
+
+    /**
+     * @notice The function to get the owner of a token
+     * @param tokenId_ the token to get the owner of
+     * @return address of an owner or `address(0)` if token does not exist
+     */
+    function ownerOf(uint256 tokenId_) public view override returns (address) {
+        return _tokenOwners[tokenId_];
+    }
+
+    /**
+     * @notice The function to get the base URI of all the tokens
+     * @return the base URI
+     */
+    function baseURI() public view virtual override returns (string memory) {
+        return _baseURI;
+    }
+
+    /**
+     * @notice The function to get the token URI.
+     *
+     * - If individual token URI is set, it gets returned.
+     * - Otherwise if base URI is set, the concatenation of base URI and token URI gets returned.
+     * - Otherwise `""` gets returned
+     *
+     * @param tokenId_ the token to get the URI of
+     * @return the URI of the token
+     */
+    function tokenURI(uint256 tokenId_) public view virtual override returns (string memory) {
```

Comment:
Should we process a branch when the `tokenId_` does not exist?

Reply:
Reverting view functions is usually not good